File: hotelBookingSystemBack/GateawayService/src/main/java/com/bmstu/gateawayservice/requests/Hotel.java
package com.bmstu.gateawayservice.requests;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.util.Collection;

@Getter
@Setter
@ToString
@RequiredArgsConstructor
public class Hotel {
    private Long id;
    private String name;
    private String country;
    private String city;
    private String address;
    private String description;
    private int numberAvailableRooms;
    private int numberRooms;
    private String ownerLogin;
    private Double price;
}

File: frontend/src/types/hotel.ts
export interface HotelsResult {
  response: Hotel[];
  // success: boolean;
}

export interface Hotel {
  uuid?: string;
  id: string;
  name: string;
  country: string;
  city: string;
  address: string;
  description: string;
  numberAvailableRooms: number;
  numberRooms: number;
  ownerLogin: string;
  price: number;
}

export interface HotelsFilter {
  country: string;
  city: string;
  dateFrom: string;
  dateTo: string;
}

export interface HotelCreate {
  name : string;
  country : string;
  city : string;
  address : string;
  description : string;
  numberAvailableRooms : number;
  numberRooms : number;
  ownerLogin : string;
  price : number;
}


export interface BookingParams {
  refundAmount: string;
  description: string;
}
